polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -1282,4 +1273,49 @@ impl<Config: config::Config> XcmExecutor<Config> {
 				}),
 		}
 	}
+
+
+	/// Deposit `to_deposit` assets to `beneficiary`, without giving up on the first (transient) error, and
+	/// retrying once just in case one of the subsequently deposited assets satisfy some requirement.
+	///
+	/// Most common transient error is: `beneficiary` account does not yet exist and the first asset(s) in
+	/// the (sorted) list does not satisfy ED, but a subsequent one in the list does.
+	///
+	/// This function can write into storage and also return an error at the same time, it should always be
+	/// called within a transactional process.
+	fn deposit_assets_with_retry(
+		&mut self,
+		to_deposit: &AssetsInHolding,
+		beneficiary: &Location,
+	) -> Result<(), XcmError> {
+		let mut failed_deposits = Vec::with_capacity(to_deposit.len());
+		let mut first_error = None;
+
+		for asset in to_deposit.assets_iter() {
+			let asset_result =
+				Config::AssetTransactor::deposit_asset(&asset, &beneficiary, Some(&self.context));
+			// if deposit failed for asset, mark it for retry after depositing the others.
+			if asset_result.is_err() {
+				if first_error.is_none() {
+					first_error = Some(asset_result);
+				}
+				failed_deposits.push(asset);
+			}
+		}
+
+		if failed_deposits.len() == to_deposit.len() && first_error.is_some() {
+			// all deposits failed, in this case return the first error as there is no point in retrying.
+			return first_error.unwrap();
+		}
```

Comment:
Clippy is complaining you are checking first_error.is_some() and later unwrapping.

You should do if let Some(error) = first_error {} instead